libr/core/cmd_search.c

```diff
@@ -2192,6 +2192,9 @@ static void do_ref_search(RCore *core, ut64 addr,ut64 from, ut64 to, struct sear
 			if (from <= ref->addr && to >= ref->addr) {
 				r_cons_printf ("%s 0x%" PFMT64x " [%s] %s\n",
 						buf_fcn, ref->addr, r_anal_ref_type_tostring (ref->type), str);
+
+				r_flag_set (core->flags, "hit.prelude", ref->addr, strlen ("hit.prelude"));
```

Comment:
there are several wrong things in this line:

* why hit.prelude? /r have nothing to do with preludes, its about references
* this codepath is executed many times, one for each reference found, your pr is overwriting the same flag over and over
* why the flag size is the length of a string? thats unrelated to the reference it points to
* xrefs are stored as xrefs, thats why no flags are set when using /r

Reply:
the 'r' in /r stands for reference, not for research